src/SFA.DAS.EmployerFinance.AcceptanceTests/TestRepositories/TestTransactionRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using Dapper;
+using SFA.DAS.EmployerFinance.AcceptanceTests.Extensions;
+using SFA.DAS.EmployerFinance.Configuration;
+using SFA.DAS.EmployerFinance.Data;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Sql.Client;
+
+namespace SFA.DAS.EmployerFinance.AcceptanceTests.TestRepositories
+{
+    public class TestTransactionRepository : BaseRepository, ITestTransactionRepository
+    {
+        private readonly Lazy<EmployerFinanceDbContext> _employerFinanceDbContext;
+
+        public TestTransactionRepository(LevyDeclarationProviderConfiguration configuration,
+            ILog logger, Lazy<EmployerFinanceDbContext> employerFinanceDbContext)
+            : base(configuration.DatabaseConnectionString, logger)
+        {
+            _employerFinanceDbContext = employerFinanceDbContext;
+        }
+
+        public async Task<int> GetNextAccountId()
+        {
+            var result = await _employerFinanceDbContext.Value.Database.Connection.QueryFirstAsync<int>(
+                sql: "SELECT COALESCE(MAX(AccountId), 0) FROM [employer_financial].LevyDeclaration",
```

Comment:
I'm assuming this will never get called on tests that run asynchronously?

Reply:
Refactored repository slightly